sys/posix/semaphore/posix_semaphore.c

```diff
@@ -0,0 +1,71 @@
+/**
+ * Semaphore implemenation
+ *
+ * Copyright (C) 2013 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ * @ingroup posix
+ *
+ * @{
+ *
+ * @file
+ * @brief   Implementation of semaphores with priority queues
+ *
+ * @author  Christian Mehlis <mehlis@inf.fu-berlin.de>
+ * @author  René Kijewski <kijewski@inf.fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <errno.h>
+#include <inttypes.h>
+
+#include "irq.h"
+#include "sched.h"
+#include "sem.h"
+#include "tcb.h"
+#include "timex.h"
+#include "thread.h"
+#include "vtimer.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#include "semaphore.h"
+
+#define USEC_IN_NS  (1000)
```

Comment:
it would make sense to put this macro inside timex.h next to SEC_IN_USEC (even though timex_t does not support ns resolution). Any opinions?